src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -858,6 +861,59 @@ export abstract class _MatAutocompleteTriggerBase
     // but the behvior isn't exactly the same and it ends up breaking some internal tests.
     overlayRef.outsidePointerEvents().subscribe();
   }
+
+  private _trackedModals = new Set<Element>();
+
+  /**
+   * Some browsers won't expose the accessibility node of the listbox overlay if there is an
+   * `aria-modal` and the overlay element is outside of it. This method works around the issue by
+   * pointing the `aria-owns` of all modals to the overlay element.
+   *
+   * While aria-owns is not required for the ARIA 1.2 `role="combobox"` interaction pattern,
+   * it fixes an issue with VoiceOver when the autocomplete appears inside of an `aria-model="true"`
+   * element (e.g. a dialog). Without this `aria-owns`, the `aria-modal` on a dialog prevents
+   * VoiceOver from "seeing" the autocomplete's listbox overlay for aria-activedescendant.
+   * Using `aria-owns` re-parents the autocomplete overlay so that it works again.
+   * See https://github.com/angular/components/issues/20694
+   */
+  private _exposeToModals() {
+    // TODO(http://github.com/angular/components/issues/26853): consider de-duplicating this with
+    // the `LiveAnnouncer` and any other usages.
+    const id = this.autocomplete.id;
+    const modals = this._document.querySelectorAll(
+      'body > .cdk-overlay-container [aria-modal="true"]',
+    );
+
+    for (let i = 0; i < modals.length; i++) {
+      const modal = modals[i];
+      const ariaOwns = modal.getAttribute('aria-owns');
+      this._trackedModals.add(modal);
+
+      if (!ariaOwns) {
+        modal.setAttribute('aria-owns', id);
+      } else if (ariaOwns.indexOf(id) === -1) {
+        modal.setAttribute('aria-owns', ariaOwns + ' ' + id);
+      }
```

Comment:
```suggestion
      modal.setAttribute(`${modalAriaOwns || ''} ${panelId}`.trim());
```
should be able to simplify this a bit

Reply:
I think that this will duplicate the `panelId` if it's already in there.

Reply:
Good catch, the whole function should back off if the ID is already there

Reply:
I simplified this method to use `addAriaReferencedId` from the CDK. That function will back off if the ID is already there.